server/src/test/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregatorTests.java

```diff
@@ -407,6 +415,44 @@ public void testOverlappingRanges() throws IOException {
         }, new NumberFieldMapper.NumberFieldType(NUMBER_FIELD_NAME, NumberFieldMapper.NumberType.INTEGER));
     }
 
+    /**
+     * If the top level query is a runtime field we should still use
+     * {@link RangeAggregator.FromFilters} because we expect it'll still be faster
+     * that the normal aggregator, even though running the script for the runtime
+     * field is quite a bit more expensive than a regular query. The thing is, we
+     * won't be executing the script more times than we would if it were just at
+     * the top level.
+     */
+    public void testRuntimeFieldTopLevelQueryStillOptimized() throws IOException {
+        long totalDocs = (long) RangeAggregator.DOCS_PER_RANGE_TO_USE_FILTERS * 4;
+        SearchLookup lookup = new SearchLookup(s -> null, (ft, l) -> null);
+        StringFieldScript.LeafFactory scriptFactory = ctx -> new StringFieldScript("dummy", Map.of(), lookup, ctx) {
+            @Override
+            public void execute() {
+                emit("cat");
+            }
+        };
+        Query query = new StringScriptFieldTermQuery(new Script("dummy"), scriptFactory, "dummy", "cat", false);
+        debugTestCase(new RangeAggregationBuilder("r").field(NUMBER_FIELD_NAME).addRange(0, 1).addRange(1, 2).addRange(2, 3), query, iw -> {
```

Comment:
Nit - I think the formatting here isn't following the new standard; Can you just run the auto-formatter on this, please?

Reply:
Surprisingly, this is what the standard looks like. If I had to guess the lambdas are letting this bunch up somehow.

Reply:
I stand corrected.  Thanks for checking!